specs/DesignDiscussions/InputLabels.md

```diff
@@ -0,0 +1,279 @@
+    ],
+    "actions": [
+        {
+            "type": "Action.Submit",
+            "title": "Submit"
+        }
+    ]
+}
+```
+
+![img](assets/InputLabels/Placeholders.png)
+
+Possible Options:
+1. Card author can determine if the label should be put in the placeholder text (`"useLabelAsPlaceholder": true`, `"labelPosition": "Placeholder"`)
+    - What do we do if they set a non-label placeholder and set useLabelAsPlaceholder to true? (spec who wins - placeholder probably)
+2. Host decides via host-config (`"labelPosition": "Placeholder"`).
+    - Same issue as above if an explicit placeholder is set by the card author. On option would be to only respect this if no inputs have placeholders (precedent from button icon placement). 
+3. Do nothing (in the schema)
+    - Card author can set placeholder instead of label if they want a placeholder to be the label. Undercuts our ability to reason over the card (we don’t know if the placeholder text is being used as the input’s label) but I’m not sure that has any practical negative consequences:
+        - Placeholder will serve as accessibility name in the absence of a label
+        - Are there responsiveness scenarios where we’d want to pull the label out of the placeholder to expand the card? (I’d guess not?)
+        - We don’t have a good place to put the “(optional)” indicator in this scenario. That said, the same will be true of any input that doesn’t use the features spec’d in this document to associate the labels.
+
+
+4. Use labels as placeholders only as part of a future responsiveness feature (if there’s not enough space, and there are no placeholders, consolidate labels into placeholders). Could be combined with 1, 2, or 3.
+
+## Label Position
+As alluded to above, we should consider how the label position is determined. One of the goals of this feature is to allow us to someday intelligently reflow inputs, so it may be the case that we want to maintain control of label positioning. Today we would always draw the label above the input, but in the future we may put it to the side if our rendering space is wide enough, or push it into placeholders if our space is constrained.
+
+That said, it may be desirable let either the card author or the host configure the position of the label. The tradeoff is more layout power to the card author, and less flexibility to us to reflow in the future. Because of this, perhaps we should not introduce this with v1 of this feature, and consider it only if we get customer feedback or requests to allow positioning of labels.
+
+```json
+{
+	"type": "Input.Text",
+	"id": "nameInput",
+	"label": "Please enter your name:",
+	"labelPosition": "Above",
+	"placeholder": "Last, First"
+}
+```
+![img](assets/InputLabels/LabelPositionAbove.png)
+
+```json
+{
+	"type": "Input.Text",
+	"id": "nameInput",
+	"label": "Please enter your name:",
+	"labelPosition": "Side",
+	"placeholder": "Last, First"
+}
+```
+![img](assets/InputLabels/LabelPositionSide.png)
```

Comment:
>LabelPositionSide [](start = 26, length = 17)

Matt: We would also need a way to group inputs so that their labels have the same width and the inputs line up.

Reply:
Added a note about this to the doc.

---
In reply to: [347042572](https://github.com/microsoft/AdaptiveCards/pull/3618#discussion_r347042572) [](ancestors = 347042572)